dmscripts/send_stats_to_performance_platform.py

```diff
@@ -0,0 +1,163 @@
+import json
+
+import backoff
+import base64
+import requests
+
+from datetime import datetime, timedelta
+import dmapiclient
+
+from dmscripts.helpers import logging_helpers
+from dmscripts.helpers.logging_helpers import logging
+
+HOURLY_TIME_FORMAT = '%Y-%m-%dT%H:00:00+00:00'  # On the hour exactly
+DAILY_TIME_FORMAT = '%Y-%m-%dT00:00:00+00:00'  # Midnight
+PERFORMANCE_PLATFORM_URLS = {
+    "day": {
+        "stage": "https://www.performance.service.gov.uk/data/gcloud/applications-by-stage",
+        "lot": "https://www.performance.service.gov.uk/data/gcloud/applications-by-lot",
+    },
+    "hour": {
+        "stage": "https://www.performance.service.gov.uk/data/gcloud/applications-by-stage-realtime",
+        "lot": "https://www.performance.service.gov.uk/data/gcloud/applications-by-lot-realtime",
+    },
+}
+
+logger = logging_helpers.configure_logger({'dmapiclient': logging.WARNING})
+
+
+def _format_statistics(stats, category, groupings):
+    return _label_and_count(stats[category], groupings)
+
+
+def _label_and_count(stats, groupings):
+    data = {
+        label: _sum_counts(stats, filters)
+        for label, filters in groupings.items()
+    }
+    return data
+
+
+def _sum_counts(stats, filter_by=None, sum_by='count'):
+    return sum(
+        statistic[sum_by] for statistic in stats
+        if not filter_by or all(
+            _find(statistic.get(key), value)
+            for key, value in filter_by.items()
+        )
+    )
+
+
+def _find(statistic, filter_value):
+    if isinstance(filter_value, list):
+        return statistic in filter_value
+    else:
+        return statistic == filter_value
+
+
+def _generate_id(timestamp, period, data_type, data_item):
+    # Instructions from Performance Platform:
+    # _id should be a unique url-friendly, base64-encoded, UTF8 encoded concatenation identifier, formed from:
+    # _timestamp, service (= gcloud), period (= day or hour), dataType (= applications-by-stage/lot), stage/lot
+    id_bytes = '{}-gcloud-{}-{}-{}'.format(timestamp, period, data_type, data_item).encode('utf-8')
```

Comment:
We only have one dataset that exists on performance platform at the moment. If we want new non- `gcloud` datasets they will need to set them up for us, and so it felt like using the name  in here is OK.   If it's parameterised then people might get the idea they could send a different value and expect it to work.